bonding-farm/contracts/AutoStakingBond.sol

```diff
@@ -0,0 +1,888 @@
+
+        /**
+            principal is transferred
+         */
+        IERC20Upgradeable(_principal).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _amount
+        );
+        totalPrincipals[_principal] += _amount;
+
+        /**
+            create bond info
+        */
+        uint256 depositId = depositIdGenerator.current();
+        depositIdGenerator.increment();
+        // depositor info is stored
+        bondInfo[depositId] = Bond({
+            depositId: depositId,
+            principal: _principal,
+            amount: _amount,
+            payout: payout,
+            vesting: _lockingPeriod,
+            lastBlockAt: block.timestamp,
+            pricePaid: priceInUSD,
+            depositor: msg.sender,
+            stake: _stake,
+            fnftId: 0
+        });
+
+        /**
+            auto staking payout
+        */
+        if (_stake) {
+            lockFarm.stake(payout, _lockingPeriod);
+            bondInfo[depositId].fnftId = fnft.tokenByIndex(
+                fnft.totalSupply() - 1
+            );
+        }
+
+        ownedDeposits[msg.sender][depositCounts[msg.sender]] = depositId;
+        depositIndexes[depositId] = depositCounts[msg.sender];
+        depositCounts[msg.sender] = depositCounts[msg.sender] + 1;
+
+        // indexed events are emitted
+        emit BondCreated(
+            depositId,
+            _principal,
+            _amount,
+            _stake,
+            payout,
+            block.timestamp + _lockingPeriod,
+            priceInUSD
+        );
+
+        processFee(_principal, _amount); // distribute fee
+
+        return payout;
+    }
+
+    /**
+     *  @notice redeem bond for user
+     *  @param _depositId uint
+     *  @return uint
+     */
+    function redeem(uint256 _depositId)
+        external
+        whenNotPaused
+        returns (uint256)
+    {
+        Bond memory info = bondInfo[_depositId];
+        address _recipient = info.depositor;
+        require(msg.sender == _recipient, 'Cant redeem others bond');
+
+        uint256 percentVested = percentVestedFor(_depositId); // (blocks since last interaction / vesting term remaining)
+
+        require(percentVested >= ONEinBPS, 'Not fully vested');
+
+        delete bondInfo[_depositId]; // delete user info
+
+        totalRemainingPayout -= info.payout; // total remaining payout is decreased
+
+        /**
+            withdraw and claim from lock farm
+         */
+        if (info.stake) {
+            IERC20Upgradeable farmRewardToken = IERC20Upgradeable(
+                lockFarm.rewardToken()
+            );
+            uint256 before = farmRewardToken.balanceOf(address(this));
+            lockFarm.withdraw(info.fnftId); // withdraw from lock farm
+            uint256 claimedAmount = farmRewardToken.balanceOf(address(this)) -
+                before;
+
+            if (stakingFee) {
+                uint256 fee = (claimedAmount * feeBps) / ONEinBPS;
```

Comment:
@bejaxer i think we need a new variable to track the autostaking fee instead of using the feeBps. 
Also, once the fee is calculated, we probably want to use the logic in func processFee (line 659-675) to distribute the auto-staking fee to fee recipients.
perhaps, refactoring the processFee to handle both cases?